tensorflow_probability/python/mcmc/internal/util.py

```diff
@@ -332,8 +332,10 @@ def smart_for_loop(loop_num_iter, body_fn, initial_loop_vars,
   """
   with tf.name_scope(name or 'smart_for_loop'):
     loop_num_iter_ = tf.get_static_value(loop_num_iter)
-    if (loop_num_iter_ is None or tf.executing_eagerly() or
```

Comment:
Please add a comment explaining the loop_num_iter_ > 1 check.  The existence of Issue #1033 helps make the comment concise, but it should be there.

Reply:
A comment around line 350 that by that point `loop_num_iter_` is no more than 1 would also help make this code clearer.

Reply:
Ah right. My bad -- forgot about that comment